pyuvdata/uvcal/uvcal.py

```diff
@@ -4566,6 +5059,359 @@ def read_fhd_cal(
+                run_check=run_check,
+                check_extra=check_extra,
+                run_check_acceptability=run_check_acceptability,
+                # file-type specific parameters
+                # FHD
+                obs_file=obs_file_use,
+                layout_file=layout_file_use,
+                settings_file=settings_file_use,
+                raw=raw,
+                extra_history=extra_history,
+            )
+            uv_list = []
+            for ind, file in enumerate(filename[1:]):
+                if file_type == "fhd":
+                    file_index = ind + 1
+                    obs_file_use = obs_file[file_index]
+                    if layout_file is not None:
+                        layout_file_use = layout_file[file_index]
+                    if settings_file is not None:
+                        settings_file_use = settings_file[file_index]
+
+                uvcal2 = UVCal()
+                uvcal2.read(
+                    file,
+                    read_data=read_data,
+                    file_type=file_type,
+                    use_future_array_shapes=use_future_array_shapes,
+                    # checking parameters
+                    run_check=run_check,
+                    check_extra=check_extra,
+                    run_check_acceptability=run_check_acceptability,
+                    # file-type specific parameters
+                    # FHD
+                    obs_file=obs_file_use,
+                    layout_file=layout_file_use,
+                    settings_file=settings_file_use,
+                    raw=raw,
+                    extra_history=extra_history,
+                )
+                uv_list.append(uvcal2)
+            # Concatenate once at end
+            if axis is not None:
+                # fast_concat to operates on lists
+                self.fast_concat(
+                    uv_list,
+                    axis,
+                    run_check=run_check,
+                    check_extra=check_extra,
+                    run_check_acceptability=run_check_acceptability,
+                    inplace=True,
+                )
+            else:
+                # Too much work to rewrite __add__ to operate on lists
+                # of files, so instead doing a binary tree merge
+                uv_list = [self] + uv_list
+                while len(uv_list) > 1:
+                    for uv1, uv2 in zip(uv_list[0::2], uv_list[1::2]):
+                        uv1.__iadd__(
+                            uv2,
+                            run_check=run_check,
+                            check_extra=check_extra,
+                            run_check_acceptability=run_check_acceptability,
+                        )
+                    uv_list = uv_list[0::2]
+                # Because self was at the beginning of the list,
+                # everything is merged into it at the end of this loop
+        else:
+            if file_type == "calfits":
```

Comment:
I know we don't support any other types, but do we want an `else` block letting the user know what the viable choices are for `file_type`? Otherwise I think it'll just silently do nothing...

Reply:
good idea.